### PR TITLE
VIPER for WipeDatabaseViewController

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

fix for https://github.com/wireapp/wire-ios/pull/4495/files#r483689786
Apple VIPER for `WipeDatabaseViewController`

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabasePresenter.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+final class WipeDatabasePresenter {
+    private weak var userInterface: WipeDatabaseUserInterface?
+    private var interactorInput: WipeDatabaseInteractorInput
+
+    convenience init(userInterface: WipeDatabaseUserInterface) {
+        let interactor = WipeDatabaseInteractor()
+        self.init(userInterface: userInterface, interactorInput: interactor)
+        interactor.output = self
+    }
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabasePresenter.swift:28]: I'd say this is the job of the Wireframe / Router (not sure which terminology we decided to go for)
@marcomaddy1982 what do you think ?

File: Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabaseViewController.swift [hunk 2]
```diff
@@ -16,11 +16,38 @@
...
+}
+
+extension WipeDatabaseViewController: WipeDatabaseUserInterface {
+    func presentWipeCompletionViewController() {
+        let wipeCompletionViewController = WipeCompletionViewController()
+        wipeCompletionViewController.modalPresentationStyle = .fullScreen
+        
+        AppDelegate.shared.notificationsWindow?.isHidden = false
+        present(wipeCompletionViewController, animated: true)
+    }
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabaseViewController.swift:32]: same comment regarding routing

File: Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabasePresenter.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabasePresenter.swift:28]: The setup can be put in wireframe. Do you have a standart example?

Comment (David-Henner) [reply Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabasePresenter.swift:28]: I'd create a wireframe class and add this method to present the wipe database module

```
func presentWipeDatabaseModule(from presentingViewController: UIViewController) {
   let interactor = WipeDatabaseInteractor()
   let presenter = WipeDatabasePresenter()
   let viewController = WipeDatabaseViewController()

   viewController.presenter = presenter
   presenter.userInterface = viewController
   presenter.interactorInput = interactor
   interactor.output = presenter
   
   presentingViewController.present(viewController, animated: true)
}
```

File: Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabaseWireframe.swift [hunk 3]
```diff
@@ -19,7 +19,7 @@
 import UIKit
 
 final class WipeDatabaseWireframe {
-    func presentWipeDatabaseModule(from presentingViewController: UIViewController) {
+    func createWipeDatabaseModule() -> WipeDatabaseViewController {
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabaseWireframe.swift:22]: maybe use a computed property here?

**Review by David-Henner — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabaseWireframe.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Overlay/CustomAppLock/WipeDatabase/WipeDatabaseWireframe.swift:22]: It is a factory method and not providing a property of `WipeDatabaseWireframe`, it should be a method here.

**Pull request merged**